AnkiDroid/src/main/java/com/ichi2/anki/AddonsBrowser.java

```diff
@@ -0,0 +1,386 @@
+
+        reviewerAddons.setOnMenuItemClickListener(item -> {
+            getSupportActionBar().setTitle(getString(R.string.reviewer_addons));
+            listAddonsFromDir(addonType);
+            return true;
+        });
+
+        downloadDialog = new Dialog(this);
+
+        installAddon.setOnMenuItemClickListener(item -> {
+
+            downloadDialog.setCanceledOnTouchOutside(true);
+            downloadDialog.setContentView(R.layout.addon_install_from_npm);
+
+            EditText downloadEditText = downloadDialog.findViewById(R.id.addon_download_edit_text);
+            Button downloadButton = downloadDialog.findViewById(R.id.addon_download_button);
+
+            downloadButton.setOnClickListener(v -> {
+                String npmAddonName = downloadEditText.getText().toString();
+
+                // if string is:  npm i ankidroid-js-addon-progress-bar
+                if (npmAddonName.startsWith("npm i")) {
+                    npmAddonName = npmAddonName.replace("npm i", "");
+                }
+
+                // if containing space
+                npmAddonName = npmAddonName.trim();
+                npmAddonName = npmAddonName.replaceAll("\u00A0", "");
+
+                if (npmAddonName.isEmpty()) {
+                    return;
+                }
+
+                UIUtils.showThemedToast(this, getString(R.string.downloading_addon), true);
+                // download npm package for AnkiDroid as addons
+                AddonsNpmUtility.getPackageJson(npmAddonName, this);
+
+                downloadDialog.dismiss();
+
+            });
+
+            downloadDialog.show();
+            return true;
+        });
+
+        getMoreAddons.setOnMenuItemClickListener(item -> {
+            openUrl(Uri.parse(getResources().getString(R.string.ankidroid_js_addon_npm_search_url)));
+            return true;
+        });
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+
+    /*
+       list addons with valid package.json, i.e contains
+       ankidroid_js_api = 0.0.1
+       keywords='ankidroid-js-addon'
+       and non empty string.
+       Then that addon will available for enable/disable
+     */
+    public static void listAddonsFromDir(String addonType) {
+        boolean success = true;
+        if (!addonsHomeDir.exists()) {
+            success = addonsHomeDir.mkdirs();
+        }
+
+        if (success) {
+
+            // reset for new view create
+            addonsNames.clear();
+
+            File directory = new File(String.valueOf(addonsHomeDir));
+            File[] files = directory.listFiles();
+            for (File file : files) {
+                Timber.d("Addons: %s", file.getName());
+
+                // Read package.json from
+                // AnkiDroid/addons/ankidroid-addon-../package/package.json
+                File addonsPackageDir = new File(file, "package");
+                File addonsPackageJson = new File(addonsPackageDir, "package.json");
```

Comment:
I think this will work

```suggestion
                File addonsPackageJson = new File(file, "package/package.json");
```